core/unit_test/CMakeLists.txt

```diff
@@ -67,106 +67,149 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;HIP)
-      ${dir}/Test${Tag}_SubView_c02.cpp
-      ${dir}/Test${Tag}_SubView_c03.cpp
-      ${dir}/Test${Tag}_SubView_c04.cpp
-      ${dir}/Test${Tag}_SubView_c05.cpp
-      ${dir}/Test${Tag}_SubView_c06.cpp
-      ${dir}/Test${Tag}_SubView_c07.cpp
-      ${dir}/Test${Tag}_SubView_c08.cpp
-      ${dir}/Test${Tag}_SubView_c09.cpp
-      ${dir}/Test${Tag}_SubView_c10.cpp
-      ${dir}/Test${Tag}_SubView_c11.cpp
-      ${dir}/Test${Tag}_SubView_c12.cpp
-      ${dir}/Test${Tag}_SubView_c13.cpp
-      ${dir}/Test${Tag}_Team.cpp
-      ${dir}/Test${Tag}_TeamReductionScan.cpp
-      ${dir}/Test${Tag}_TeamScratch.cpp
-      ${dir}/Test${Tag}_TeamTeamSize.cpp
-      ${dir}/Test${Tag}_TeamVectorRange.cpp
-      ${dir}/Test${Tag}_UniqueToken.cpp
-      ${dir}/Test${Tag}_ViewAPI_a.cpp
-      ${dir}/Test${Tag}_ViewAPI_b.cpp
-      ${dir}/Test${Tag}_ViewAPI_c.cpp
-      ${dir}/Test${Tag}_ViewAPI_d.cpp
-      ${dir}/Test${Tag}_ViewAPI_e.cpp
-      ${dir}/Test${Tag}_ViewLayoutStrideAssignment.cpp
-      ${dir}/Test${Tag}_ViewMapping_a.cpp
-      ${dir}/Test${Tag}_ViewMapping_b.cpp
-      ${dir}/Test${Tag}_ViewMapping_subview.cpp
-      ${dir}/Test${Tag}_ViewOfClass.cpp
-      ${dir}/Test${Tag}_WorkGraph.cpp
-      ${dir}/Test${Tag}_View_64bit.cpp
-      ${dir}/Test${Tag}_ViewResize.cpp
-  )
-  SET(${Tag}_SOURCES ${${Tag}_SOURCES1} ${${Tag}_SOURCES2})
+  if(Kokkos_ENABLE_${DEVICE})
+    set(dir ${CMAKE_CURRENT_BINARY_DIR}/${dir})
+    file(MAKE_DIRECTORY ${dir})
+    # Needed to split this for Windows NVCC, since it ends up putting everything on the
+    # command line in an intermediate compilation step even if CMake generated a response
+    # file. That then exceeded the shell comand line max length.
+    set(${Tag}_SOURCES1)
+    foreach(Name
+        AtomicOperations_int
+        AtomicOperations_unsignedint
+        AtomicOperations_longint
+        AtomicOperations_unsignedlongint
+        AtomicOperations_longlongint
+        AtomicOperations_double
+        AtomicOperations_float
+        AtomicOperations_complexdouble
+        AtomicOperations_complexfloat
+        AtomicViews
+        Atomics
+        BlockSizeDeduction
+        Concepts
+        Complex
+        Crs
+        DeepCopyAlignment
+        FunctorAnalysis
+        Init
+        LocalDeepCopy
+        MDRange_a
+        MDRange_b
+        MDRange_c
+        MDRange_d
+        MDRange_e
+        MDRange_f
+        Other
+        RangePolicy
+        RangePolicyRequire
+        Reductions
+        Reducers_a
+        Reducers_b
+        Reducers_c
+        Reducers_d
+        Reductions_DeviceView
+        Scan
+        SharedAlloc
+        )
+      set(file ${dir}/Test${Tag}_${Name}.cpp)
+      file(WRITE ${file}
+          "#include <Test${Tag}_Category.hpp>\n"
+          "#include <Test${Name}.hpp>\n"
+      )
+      list(APPEND ${Tag}_SOURCES1 ${file})
+    endforeach()
+
+    SET(${Tag}_SOURCES2)
+    foreach(Name
+      SubView_a
+      SubView_b
+      TeamBasic
```

Comment:
Needed to rename `Team` to `TeamBasic` since there are multiple tests including `Team.hpp`.